src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -512,12 +518,36 @@ namespace AppInstaller::Repository
                 AICLI_LOG(Repo, Info, << "Default source requested, but no sources configured");
                 return {};
             }
-            else
+            else if(currentSources.size() == 1)
             {
-                // TODO: Create aggregate source here.  For now, just get the first in the list.
-                AICLI_LOG(Repo, Info, << "Default source requested, using first source: " << currentSources[0].Name);
+                AICLI_LOG(Repo, Info, << "Default source requested, only 1 source available, using the only source: " << currentSources[0].Name);
                 return OpenSource(currentSources[0].Name, progress);
             }
+            else
+            {
+                AICLI_LOG(Repo, Info, << "Default source requested, multiple sources available, creating aggregated source.");
+                auto aggregatedSource = std::make_shared<AggregatedSource>();
+
+                bool sourceUpdated = false;
+                for (auto& source : currentSources)
+                {
+                    AICLI_LOG(Repo, Info, << "Adding to aggregated source: " << source.Name);
+
+                    if (ShouldUpdateBeforeOpen(source))
+                    {
+                        UpdateSourceFromDetails(source, progress);

```

Comment:
>progress [](start = 56, length = 8)

At some point we are going to have to add a context callback to indicate "we are doing this action now".  This loop could potentially have the progress bar fill up 6+ times with just two sources.

Reply:
Added todo

---
In reply to: [467334631](https://github.com/microsoft/winget-cli/pull/527#discussion_r467334631) [](ancestors = 467334631)